modules/imgcodecs/src/loadsave.cpp

```diff
@@ -1032,6 +991,265 @@ bool haveImageWriter( const String& filename )
     return !encoder.empty();
 }
 
+class ImageCollection::Impl {
+public:
+    Impl() = default;
+    Impl(const std::string&  filename, int flags);
+    void init(String const& filename, int flags);
+    size_t size() const;
+    Mat at(int index);
+    Mat& operator[](int index);
+    void release(int index);
```

Comment:
I propose to rename the method to `ReleaseCache()`. Other OpenCV classes that contain `release()` method finalize the algorithm, but not clean cache. Object usage after `release()` is not expected.

Reply:
I inspired from cv::Mat. The intention of the both call is almost the same. But i am fine with the renaming of the method name.

Reply:
@asmorkalov you marked this as resolved. In the todays meeting, Vadim also agreed with you to rename the method name. I got confused why you marked this as resolved. I was going to rename it. Should I?

Reply:
My fault. I resolved wrong discussion. Renamed method looks better, thanks!